position/sync/chooseOneResults.js

```diff
@@ -0,0 +1,30 @@
+var isArray = require('isarray')
+var Position = require('../../position/sync/position')
+
+//Expects `poll` and `position` objects passed in to be of shape: 
+//{
+//  value: {
+//    content: {...},
+//    timestamp: ...
+//    author: ...
+//  }
+//}
+//
+//postions must be of the correct type ie: type checked by the caller.
+module.exports = function ({positions, poll}) {
+  return positions.reduce(function (results, position) {
+    var { positionDetails: {choice} } = Position(position)
+
+    if (choice >= poll.pollDetails.choices.length || position.value.timestamp > poll.closesAt) {
+      results.errors.invalidPositions.push(position)
```

Comment:
I think perhaps the errors should be on the poll, as I'm going to be collecting errors up there too:
```js
var poll = {
  key,
  value,
  positions,
  results,  // your work fits in here
  errors: [
    { type: 'Invalid position type', position },
    { type: 'late position', position },
    { type: 'Invalid choice', position }
  ]
```

I'm making this up... documentation driven development?

Reply:
```js
var poll = {
  key,
  value,
  positions,
  results,  // your work fits in here
  errors: [
    { type: 'INVALID_POSITION' ,message: 'Invalid position type', position },
//    { type: new TypeError('invalid position'), message: 'late position', position },
    { type, message: 'Invalid choice', position }
  ]
}
```

Reply:
Half done. Rest is on a different ticket.